lib/URL.ts

```diff
@@ -0,0 +1,38 @@
+import { ToStringable } from "./String"
+
+export type URLParameters = {
+  [key: string]: ToStringable | undefined
+}
+
+export type Endpoint = `/${string}`
+
+export const urlConstructor = (baseURL: URL, endpoint: Endpoint, query?: URLParameters, params?: URLParameters) => {
```

Comment:
```suggestion
export const urlString = (baseURL: URL, endpoint: Endpoint, query?: URLParameters, params?: URLParameters) => {
```
This doesn't really create a `URL`, so I think the latter name is preferrable.

Reply:
Thanks, fixed